### PR TITLE
[Feature] introduce plan mode and strategies

### PR DESCRIPTION
**Pull request by stephen-shelby**
## What I'm doing:
1. add plan mode session variables
2. add iceberg distributed plan strategies

Fixes #issue
https://github.com/StarRocks/starrocks/issues/43460

## What type of PR is this:

- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/iceberg/StarRocksIcebergTableScan.java [hunk 1]
```diff
@@ -321,6 +323,45 @@ private FileScanTask toFileScanTask(DataFile dataFile) {
...
+    }
+
+    // TODO(stephen): add more strategies
+    private boolean shouldPlanLocally(List<ManifestFile> manifests) {
+        switch (planMode) {
+            case LOCAL:
+                return true;
+
+            case DISTRIBUTED:
+                return manifests.isEmpty();
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/org/apache/iceberg/StarRocksIcebergTableScan.java:337]: just return false?

File: fe/fe-core/src/main/java/org/apache/iceberg/StarRocksIcebergTableScan.java [hunk 2]
```diff
@@ -321,6 +323,45 @@ private FileScanTask toFileScanTask(DataFile dataFile) {
...
+        return workers.stream().mapToInt(ComputeNode::getCpuCores).sum();
+    }
+
+    private long totalSize(List<ManifestFile> manifests) {
+        return manifests.stream().mapToLong(ManifestFile::length).sum();
+    }
+
+    private boolean mayHaveEqualityDeletes(Snapshot snapshot) {
+        String count = snapshot.summary().get(SnapshotSummary.TOTAL_EQ_DELETES_PROP);
+        return count == null || !count.equals("0");
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/org/apache/iceberg/StarRocksIcebergTableScan.java:362]: count == null may has EqualityDeletes?

File: fe/fe-core/src/main/java/org/apache/iceberg/StarRocksIcebergTableScan.java (hunk 1 above)
Comment (stephen-shelby) [reply fe/fe-core/src/main/java/org/apache/iceberg/StarRocksIcebergTableScan.java:337]: if there are no manifest, we should use local mode and short circuit

File: fe/fe-core/src/main/java/org/apache/iceberg/StarRocksIcebergTableScan.java (hunk 2 above)
Comment (stephen-shelby) [reply fe/fe-core/src/main/java/org/apache/iceberg/StarRocksIcebergTableScan.java:362]: yes

**Review by Youngwb — APPROVED** [review]

**Review by imay — APPROVED** [review]

**Pull request merged**